### PR TITLE
Clarify CRE spec and note which instructions depend on it

### PR DESCRIPTION
**Pull request by tariqkurd-repo**
The CRE spec was ambiguous as highlighted in https://github.com/riscv/riscv-cheri/issues/218
This is an attempt to fix it and show which instructions need CRE=1 to execute

### CODE REVIEW

**Comment by tariqkurd-repo** [comment]
The interesting change is in riscv-legacy-integration.adoc
The large number of others are adding the exception in to the relevant instruction pages

File: src/insns/sh4add_32bit.adoc [hunk 1]
```diff
@@ -35,6 +35,9 @@ Increment the address field of `cs2` by `rs1` shifted left by 4 bit positions. C
 Legacy Mode Description::
 Increment `rs2` by `rs1` shifted left by 4 bit positions.
 
+Exceptions::
+include::require_cre.adoc[]
```
Comment (arichardson) [inline src/insns/sh4add_32bit.adoc:39]: Do we really need CRE for this instruction? I would assume it works fine in legacy mode?

File: src/insns/sh4adduw_32bit.adoc [hunk 2]
```diff
@@ -28,6 +28,9 @@ Encoding::
 Capability Mode Description::
 Increment the address field of `cs2` by the unsigned word in `rs1` shifted left by 4 bit positions. Clear the tag if the resulting capability is unrepresentable or `cs2` is sealed.
 
+Exceptions::
+include::require_cre.adoc[]
```
Comment (arichardson) [inline src/insns/sh4adduw_32bit.adoc:32]: Same here.

**Review by arichardson — APPROVED** [review]
This looks good, but I'm not sure whether the restriction is needed for the shadd in legacy mode. I'm leaning no, but maybe it makes sense for simplicity.

**Comment by tariqkurd-repo** [comment]
> This looks good, but I'm not sure whether the restriction is needed for the shadd in legacy mode. I'm leaning no, but maybe it makes sense for simplicity.

It's a choice. It's easier to define "all instructions in the cheri extensions depend on CRE".
The alternative, which is quite reasonable, is to make another mini-extension for SH4ADD.*, and then treat them the same way as SH[123]ADD.*, as these are instructions which could be used on a non_CHERI core.

I'd prefer one of those two options, rather then have them in a CHERI extension but exclude them, to keep the definition clean.

maybe a new mini-extension is the way to go.

File: src/insns/sh4adduw_32bit.adoc (hunk 2 above)
Comment (tariqkurd-repo) [reply src/insns/sh4adduw_32bit.adoc]: fixed - it's now allowed

**Pull request merged**